### ISSUE TITLE
container image build fails when env variable contains comma

### ISSUE THREAD

**Issue by Weiss91** [bazelbuild/rules_docker#555]
I encountered a bug where the container build fails if an environment variable declared in the `env` parameter contains a comma.
E.g.:
```
container_image(
    name = "image_test_bug",
    base = ":layer_test_bug",
    cmd = ["run"],
    env = {
        "BUGGY_ENV_VAR": "This does not work with a comma ,",
    },
    ports = ["12345"],
    stamp = True,
    visibility = ["//visibility:public"],
)
```

Minimal working example: 
https://github.com/Weiss91/bazelDockerRulesCommaBug

To see the bug, run `bazel build image_test_bug`

output: 
```$ bazel build image_test_bug
INFO: Analysed target //:image_test_bug (1 packages loaded).
INFO: Found 1 target...
ERROR: /mnt/c/Users/matthias.weiss/go/src/github.com/weiss91/bazelDockerEnvBug/BUILD.bazel:21:1: ImageConfig image_test_bug.0.config failed (Exit 1)
Traceback (most recent call last):
  File "/home/matthias/.cache/bazel/_bazel_matthias/876a68a91054abbeffa7e92ef6acf28e/sandbox/processwrapper-sandbox/1/execroot/test_env/bazel-out/host/bin/external/io_bazel_rules_docker/container/create_image_config.runfiles/io_bazel_rules_docker/container/create_image_config.py", line 208, in <module>
    main()
  File "/home/matthias/.cache/bazel/_bazel_matthias/876a68a91054abbeffa7e92ef6acf28e/sandbox/processwrapper-sandbox/1/execroot/test_env/bazel-out/host/bin/external/io_bazel_rules_docker/container/create_image_config.runfiles/io_bazel_rules_docker/container/create_image_config.py", line 184, in main
    for (k, v) in six.iteritems(KeyValueToDict(args.env))
  File "/home/matthias/.cache/bazel/_bazel_matthias/876a68a91054abbeffa7e92ef6acf28e/sandbox/processwrapper-sandbox/1/execroot/test_env/bazel-out/host/bin/external/io_bazel_rules_docker/container/create_image_config.runfiles/io_bazel_rules_docker/container/create_image_config.py", line 95, in KeyValueToDict
    (k, v) = kv.split('=', 1)
ValueError: need more than 1 value to unpack
Target //:image_test_bug failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2.517s, Critical Path: 0.79s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

**Comment by nlopezgi** [comment]
Thanks for reporting this issue. Its likely we had never tried this use case.  Likely the issue is related to how env variables are being processed here: https://github.com/bazelbuild/rules_docker/blob/master/container/create_image_config.py#L184 and split up here: https://github.com/bazelbuild/rules_docker/blob/master/container/create_image_config.py#L95. It would be awesome if you could try sending a PR with a fix that works for your use case! Otherwise we'll try to fix this in the upcoming weeks.

**Comment by Xjs** [comment]
Disclaimer: I work with @Weiss91 and we talked about this. They were running the build on a WSL host. I can confirm the error occurs on a »real« Linux as well, in exactly the same way. Can you give us a pointer as to where `create_image_config.py` is called? The problem is likely in the processing of its command line arguments.

**Comment by nlopezgi** [comment]
hi @Xjs , thanks for the additional details,
`create_image_config.py` is called here:
https://github.com/bazelbuild/rules_docker/blob/master/container/image.bzl#L195
Likely the error is, as you mention, in how we process args; I'd be very happy to help review a PR adding a test exercising your use case and fixing the bug!

**Comment by alexbumbacea** [comment]
I think this can be closed as it was solved in #577
